Dockerfile

```diff
@@ -1,15 +1,37 @@
-FROM python:3.9
+FROM python:3.9-alpine as base
+
+ENV PYTHONFAULTHANDLER=1 \
+    PYTHONHASHSEED=random \
+    PYTHONUNBUFFERED=1
+
+ENV VENV_PATH="/venv"
+
+ENV PATH="$VENV_PATH/bin:$PATH"
 
 WORKDIR /app
 
-RUN apt-get update && apt-get upgrade -y
+FROM base as builder
 
-RUN pip install --upgrade pip wheel setuptools poetry
+ENV PIP_DEFAULT_TIMEOUT=100 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1 \
+    PIP_NO_CACHE_DIR=1 \
+    POETRY_VERSION=1.0.5
 
-ENV POETRY_VIRTUALENVS_IN_PROJECT true
+ENV CRYPTOGRAPHY_DONT_BUILD_RUST=1
+RUN apk add --no-cache gcc libffi-dev musl-dev postgresql-dev postgresql-dev musl-dev zlib zlib-dev jpeg-dev
+RUN pip install "poetry==$POETRY_VERSION"
+RUN python -m venv /venv
+
+COPY pyproject.toml poetry.lock ./
+RUN poetry export -f requirements.txt | /venv/bin/pip install -r /dev/stdin
```

Comment:
Can you please explain why is this required?

`poetry build` itself is sufficient, I think...

Reply:
not fully aware about poetry , but what its doing installing poetry dependend package in virtualenv. which may be not reinstalled during poetry build. Will check

Reply:
yes --not required